src/software/ai/passing/cost_function.cpp

```diff
@@ -340,8 +377,47 @@ double calculateProximityRisk(const Point& point, const Team& enemy_team,
     return point_enemy_proximity_risk;
 }
 
+double ratePasserPointForKeepAway(const Pass& pass, const Team& enemy_team)
+{
+    // the default values for these passing parameters
+    // TODO: cleanup passing parameters as part of #1987
+    static constexpr double PASSER_ENEMY_PROXIMITY_IMPORTANCE   = 1.5;
+    static constexpr double RECEIVER_ENEMY_PROXIMITY_IMPORTANCE = 0.;
+    static const auto ENEMY_REACTION_TIME = Duration::fromSeconds(0);
+
+    return ratePassEnemyRisk(enemy_team, pass, ENEMY_REACTION_TIME,
+                             RECEIVER_ENEMY_PROXIMITY_IMPORTANCE) *
+           (1 - calculateProximityRisk(pass.passerPoint(), enemy_team,
+                                       PASSER_ENEMY_PROXIMITY_IMPORTANCE));
+}
+
+// TODO (NIMA): Visualize best pass point (i.e. keep away point
+double ratePasserPosition(const World& world, const Pass& pass,
+                          const Rectangle& dribbling_bounds)
+{
+    static constexpr auto KEEPAWAY_SEARCH_CIRCLE_RADIUS = 0.5;
+
+    // the width of both the field boundary sigmoid and the circular search region sigmoid
+    static constexpr auto SIGMOID_WIDTH = 0.1;
```

Comment:
should this be constants or part of dynamic params?

Reply:
maybe not `SIGMOID_WIDTH` but `KEEPAWAY_SEARCH_CIRCLE_RADIUS`

Reply:
hmm im not sure. This should be updated with @williamckha 's new field in the World that tracks where we start dribbling.